server/pkg/sass/processor.go

```diff
@@ -0,0 +1,230 @@
+package sass
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/bep/godartsass/v2"
+	"github.com/cortezaproject/corteza/server/assets"
+	"github.com/cortezaproject/corteza/server/pkg/logger"
+	"github.com/cortezaproject/corteza/server/pkg/str"
+	"go.uber.org/zap"
+)
+
+var (
+	stylesheet           = newStylesheetCache()
+	sassVariablesPattern = regexp.MustCompile(`(\$[a-zA-Z_-]+):\s*([^;]+);`)
+)
+
+// GenerateCSS takes care of creating CSS for webapps by reading SASS content from embedded assets,
+// combining it with brandingSass and customCSS, and then transpiling it using the dart-sass compiler.
+//
+// If dart sass isn't installed on the host machine, it will default to css content from the minified-custom.css which is
+// generated from [Boostrap, bootstrap-vue and custom variables sass content].
+// If dart isn't installed on the host machine, customCustom css will continue to function, but without sass support.
+//
+// In case of an error, it will return default css and log out the error
+func GenerateCSS(brandingSass, customCSS, sassDirPath string) (string, error) {
+	var (
+		stringsBuilder strings.Builder
+
+		transpiler = dartSass()
+	)
+
+	//check if cache has already compiled css value and if branding-sass and custom-css have been updated
+	if !stylesheet.empty() && stylesheet.get("branding-sass") == str.HashStringSHA256(brandingSass) && stylesheet.get("custom-css") == str.HashStringSHA256(customCSS) {
```

Comment:
I'm really not a fan of you checking the values here.
You needlessly spend processing time hashing and comparing strings which could pose a bottleneck.

Why do we not move this when updating the setting in the system service package?